File: app/src/main/java/com/gago/david/myland/Adapters/ImageItemAdapter.java
package com.gago.david.myland.Adapters;

import android.graphics.PorterDuff;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;

import com.gago.david.myland.R;

import java.util.List;

public class ImageItemAdapter extends RecyclerView.Adapter<ImageItemAdapter.ViewHolder> {

    private final List<Integer> mValues;
    private final int colotTint;

    public ImageItemAdapter(List<Integer> items, int colorTint) {
        mValues = items;
        this.colotTint = colorTint;
    }

    @Override
    public ImageItemAdapter.ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext())
                .inflate(R.layout.image_item, parent, false);
        return new ImageItemAdapter.ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(final ImageItemAdapter.ViewHolder holder, int position) {
        holder.mItem = mValues.get(position);
        holder.mIdView.setImageResource(mValues.get(position));
        holder.mIdView.setColorFilter(colotTint, PorterDuff.Mode.SRC_IN);

        holder.mView.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                /*if (null != mListener) {
                    // Notify the active callbacks interface (the activity, if the
                    // fragment is attached to one) that an item has been selected.
                    mListener.onListFragmentInteraction(holder.mItem);
                }*/
            }
        });
    }

    @Override
    public int getItemCount() {
        return mValues.size();
    }

    public class ViewHolder extends RecyclerView.ViewHolder {
        public final View mView;
        public final ImageView mIdView;
        public int mItem;

        public ViewHolder(View view) {
            super(view);
            mView = view;
            mIdView = view.findViewById(R.id.image);
        }

        @Override
        public String toString() {
            return super.toString() + " '" + mIdView.toString() + "'";
        }
    }
}

File: app/src/main/java/com/gago/david/myland/adapters/ImageItemAdapter.kt
package com.gago.david.myland.adapters

import android.graphics.PorterDuff
import androidx.recyclerview.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import com.gago.david.myland.R

class ImageItemAdapter(private val mValues: List<Int>, private val colorTint: Int) : RecyclerView.Adapter<ImageItemAdapter.ViewHolder>() {
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val view = LayoutInflater.from(parent.context)
                .inflate(R.layout.image_item, parent, false)
        return ViewHolder(view)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        holder.mItem = mValues[position]
        holder.mIdView.setImageResource(mValues[position])
        holder.mIdView.setColorFilter(colorTint, PorterDuff.Mode.SRC_IN)
        holder.mView.setOnClickListener { /*if (null != mListener) {
                    // Notify the active callbacks interface (the activity, if the
                    // fragment is attached to one) that an item has been selected.
                    mListener.onListFragmentInteraction(holder.mItem);
                }*/
        }
    }

    override fun getItemCount(): Int {
        return mValues.size
    }

    inner class ViewHolder(val mView: View) : RecyclerView.ViewHolder(mView) {
        val mIdView: ImageView
        var mItem = 0
        override fun toString(): String {
            return super.toString() + " '" + mIdView.toString() + "'"
        }

        init {
            mIdView = mView.findViewById(R.id.image)
        }
    }
}